server/src/main/java/org/elasticsearch/gateway/GatewayMetaState.java

```diff
@@ -208,8 +208,8 @@ public void setCurrentTerm(long currentTerm) {
             metaStateService.writeManifestAndCleanup("current term changed", manifest);
             previousManifest = manifest;
         } catch (WriteStateException e) {
-            logger.warn("Exception occurred when setting current term", e);
-            //TODO re-throw exception
+            logger.error("Failed to set current term to {}", currentTerm);
```

Comment:
Could we log the exception and stack trace too? `logger.error(new ParameterizedMessage("Failed to set current term to {}", currentTerm), e);`